spark3/src/main/java/org/apache/iceberg/spark/SparkCatalog.java

```diff
@@ -384,8 +384,9 @@ public boolean dropNamespace(String[] namespace) throws NoSuchNamespaceException
 
   @Override
   public final void initialize(String name, CaseInsensitiveStringMap options) {
-    this.cacheEnabled = Boolean.parseBoolean(options.getOrDefault("cache-enabled", "true"));
     Catalog catalog = buildIcebergCatalog(name, options);
+    this.cacheEnabled = Boolean.parseBoolean(options.getOrDefault("cache-enabled",
+        Boolean.toString(catalog.cachingEnabled())));
```

Comment:
From my POV the catalog value needs to have precedence of the config property because if the catalog does not work well with caching, overriding that via config will still lead to unexpected behaviours.

It might be worth renaming `catalog.cachingEnabled()` to `.cachingSupported()`.

If we really want to be able to override the catalog's value via config, perhaps we could have two config properties: `cache-enabled` (used as default) and `cache-forced` (overrides the catalog's value). WDYT?

Reply:
> From my POV the catalog value needs to have precedence of the config property because if the catalog does not work well with caching, overriding that via config will still lead to unexpected behaviours.

I was thinking about this, but figured that if a user really wants to override the caching behavior, then whatever the user configures should take precedence.

Reply:
Fair enough. In this case how about naming the new method like `catalog.cachingEnabledDefault()` for the sake of clarity?